src/query/test/promql_test.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEvaluations(t *testing.T) {
+	files, err := filepath.Glob("testdata/*.test")
+	require.NoError(t, err)
+
+	for _, fn := range files {
+		test, err := newTestFromFile(t, fn)
+		require.NoError(t, err)
+
+		require.NoError(t, test.Run())
+
+		test.Close()
+	}
+}
```

Comment:
Can we call into this package instead of copying into repo?  Seems like "NewTest(...)" is exported from this package and "newTestFromFile" is very simple and we can implement that ourselves.

Reply:
Not really as I see. `Test` has a parser which could be reused if it would be exported but it's not. `cmds` also are not exported. And then `clear` `load` and `eval` commands has a custom implementation.